### ISSUE TITLE
Integrated Circuits Shader Crash

### ISSUE THREAD

**Issue by Thornstromb** [MJRLegends/Space-Astronomy-Feedback-#259]
after playing for a while, i get this message
[integratedcircuits]: Shaders not supported, disabling aura effect :(
and the screen freezes, the ui vanishes, but the game continues running.

this is a "known" error possibly caused by faulty configs, that crashes OpenGL but not the game, as described HERE (https://github.com/Haggle1996/RevolutionPack/issues/96) for a different pack. unfortunately, the ONLY way to fix it is to A: remove the mod or B: get a fix from the pack creator. 

Sidenote: in spite of haggles comment about being able to restart and keep going, restarting DOES NOT WORK, it will immediately issue the same result upon loading in, and can even do this after a full computer restart if you are very unlucky (like me).

**Comment by MJRLegends** [comment]
If it crashes i need the crash log??

**Comment by MJRLegends** [comment]
To me it seems like a issue on your side with your graphics card and its drivers

**Comment by Thornstromb** [comment]
Firstly: there IS no "Crash" log. the minecraft logger reports no crashes, and the GAME continues running, its just the screen will not continue as OpenGL has crashed. which minecraft doesnt track. the ONLY thing on the tracker is the warning report form integrated circuits, mentioned above.
Second, this problem continues even after updating my graphics drivers, and i know my graphics "card" (an onboard) is workign perfectly fine.

the link mentions its a pack-side problem interfacing with a client-side openGL error. from what i can tell, the OpenGL error would NORMALLY not cause any actual issues except maybe a skipped frame, but combined with.... SOMETHING packside it triggers the IC mod to do.... something else, which causes OpenGL to crash completely. so basically the problem starts in OpenGL (and asking for a patch on that.... isnt going to happen), proceeds into the pack, then hits a problem in the IC mod.

to repeat, there is no crash log. as far as minecraft is concerned everything is fine. its a faulty coding that causes OpenGL to crash that is the issue. none of my friends have this problem, but it is EMINENTLY repeatable as i have yet to have a run not cut short by this damn bug. the fault is intermittent, but happens eventually no matter what i do. as a workaround were going to remove the mod from the server, but im hoping a fix can be found.

**Comment by MJRLegends** [comment]
What graphics card do you have

**Comment by Thornstromb** [comment]
AMD Radeon HD 8670D onboard.

**Comment by MJRLegends** [comment]
Have you tried to update your graphics drivers and/or removing fastcraft

**Comment by Thornstromb** [comment]
already mentioned graphics driver was updated, removing fastcraft extremely difficult due to not being the server host myself. also, nobody else on the server is getting this error.

**Comment by MJRLegends** [comment]
You still can remove fastcraft on your client side and you will still be able to join servers that have it

**Comment by MJRLegends** [comment]
@Thornstromb did my comment above solve the issue?

**Comment by MJRLegends** [comment]
I will reopen this issue if a reply is made

**Comment by tokinGanja** [comment]
Appologies, Love your modpack, but it seems i am having this exact issue, and restarting MC does work sometimes, But it's only me, and disabling fastcraft works but it wont let MC run more than 1-4 Fps, is there a setting i could change or add to the cfg file to fix it, or a shader pack i could add to counter it?

**Comment by WalaoEgg0424** [comment]
@tokinGanja what graphic card are u using?

**Comment by tokinGanja** [comment]
AMD radeon hd 4850 x2

**Comment by WalaoEgg0424** [comment]
@tokinGanja maybe turn all of the settings to lowest?

**Comment by WalaoEgg0424** [comment]
@tokinGanja btw ur graphic card is hard to run the modpack for a 60fps or more. If can , buy a new graphic card so that it runs smooth on every modpack.

**Comment by WalaoEgg0424** [comment]
@tokinGanja This is your graphic card compare to my old crappy graphic card.
http://gpuboss.com/gpus/Radeon-HD-4850-X2-vs-GeForce-GT-630

**Comment by tokinGanja** [comment]
true buteven everything at the lowest i get 1-4
Fastcract makes it run but if i disable integrated circuits i get Java null pointer exception.

**Comment by WalaoEgg0424** [comment]
@tokinGanja lower all settings , play on super small windowed mode , close anything that eats gpu or ram.

**Comment by tokinGanja** [comment]
and on single player i get no issue unless i craft with this mod, i can even disable fastcraft with no issue, but i play on the devco space astronomy server.

Ok i'll shrink my window a bit more but it's already at 640x480

**Comment by WalaoEgg0424** [comment]
@tokinGanja u mean u get low fps in server when remove fastcraft , crash in server when u delete IC. In singleplayer , runs ok when removing fastcraft + IC?

**Comment by tokinGanja** [comment]
no when i remove ic it crashes on server, when i remove fastcraft it gets 1-4 fps single player and 0-2 on the server. if i enable both i get the shader error, if i disable both runs fine single player, and does not load the server, so is there a fix i can do that affects my tower independent of the server?

Also what card would you recommend

**Comment by WalaoEgg0424** [comment]
@tokinGanja my advice is buy a new graphic card. maybe @MJRLegends can help with increasing fps but its hard when it comes to your graphic card cant get high fps.

**Comment by tokinGanja** [comment]
yeah but fastcraft is the miracle there.

**Comment by WalaoEgg0424** [comment]
@tokinGanja if u want best graphic card then gtx 1070/1080 xDD but average then gtx 970 or gt 630 or higher. My nvidia gt630 can run a smooth 120 fps on this modpack.

**Comment by tokinGanja** [comment]
makes me sometimes get a full 75-150 fps when both are enabled, but i always wait to get the shader error. sometimes it never happens, and i'll start looking to buy a new one because a steady 120 fps without the risk of crashing is great.

**Comment by tokinGanja** [comment]
thanks for the advice, if you stumble across anything or MJRLegends knows of a current fix it would be fantastic though, and i'll still post the log if it happens again, might help, should i post my DXDiag too?

**Comment by WalaoEgg0424** [comment]
@tokinGanja btw how much ram u put for the modpack?

**Comment by tokinGanja** [comment]
got it set to 3.25 of 8.0

**Comment by tokinGanja** [comment]
leaves 5.75 to operations

**Comment by WalaoEgg0424** [comment]
@tokinGanja i put 4gb of 8gb ram :) , try it , if u still gets low fps , then gpu stuff maybe.

**Comment by tokinGanja** [comment]
it runs everything up to shader model 4.9 with no issue, even skyrim, diablo 3, dead space and a lot of newre games on ultra graphic settings

**Comment by tokinGanja** [comment]
i'll give it a go really quick, havent gotten the shader error so far...

**Comment by tokinGanja** [comment]
try it with fastcraft disabled?

**Comment by WalaoEgg0424** [comment]
@tokinGanja damn u sure ur graphic card is AMD radeon hd 4850 x2 , even my graphic card cant even run BO3 on ultra low graphic settings, only 20-40fps

**Comment by tokinGanja** [comment]
i use the amd overdrive setting on new games.

**Comment by tokinGanja** [comment]
integrated controlled overclock software for the twin gpu

**Comment by tokinGanja** [comment]
1 sec

**Comment by tokinGanja** [comment]
![dxdiag](https://cloud.githubusercontent.com/assets/7274548/15813581/2a9d7350-2b8e-11e6-90a5-c33e59a9b3b9.png)

**Comment by WalaoEgg0424** [comment]
@tokinGanja do u have skype? it will be more easier to solve it :)

**Comment by tokinGanja** [comment]
yes i do.

**Comment by WalaoEgg0424** [comment]
@tokinGanja yimer1821

**Comment by tokinGanja** [comment]
there u go.

**Comment by tokinGanja** [comment]
crash just happened Posting the log as .txt, The crash instance is toward the bottom.

[FTB log.txt](https://github.com/MJRLegends/Space-Astronomy-Feedback-/files/300011/FTB.log.txt)

**Comment by tokinGanja** [comment]
is there a shader mod that would fix it for an AMD card?

**Comment by tokinGanja** [comment]
Found something new so far, optifine does work client side without interference to the server. it's not a complete fix, but it does allow play-ability, drawback is a frame rate hit, i haven't re-enabled animations and everything else yet, but it seems the only setting that changes whether or not this shader compatibility error crashes OpenGL, is by using optifine's algorithm by enabling fast render.

How I did it:
Open file location for FTB Launcher>Space astronomy folder>minecraft>mods> Drag and drop OptiFine_1.7.10_HD_U_D4.jar

Launch Space astronomy>Go to Options>Video Settings>Performance Settings>Fast Render: ON
This allows Your AMD video card to shrug off The GL error insead of crashing out(At least mine)
Tried on AMD 3000 series and my AMD Ati Radeon 4850 HD x2 2gb

It causes: [integratdecircuits] Shaders not supported, disabling aura effect.
to become: A continuious reminder in chat Stating - OpenGL error @ Post render 1282: Invalid operation

I have shaders in, but will be removing them and posting an update to that result as well.

It is my hope that by what i've found admin or MJR might be able to find a more permanent fix to the issue or narrow down the root of the AMD compatibility issue in the scripting(If there is in fact an issue in it.) Best of luck, I hope I helped someone.

![ftb log 2](https://cloud.githubusercontent.com/assets/7274548/15929755/d61126f6-2e1c-11e6-8520-1d2654a41912.png)

**Comment by MJRLegends** [comment]
One i do not support optifine and i dont support any type of shaders so if theres issues with them and the mod pack then thats down to you, as for the original issue here im not sure on a fix tbh seems to be a mod issue for sure

**Comment by tokinGanja** [comment]
the issue was on it's own i fixed it from freezing with a feature of optifine, shaders do it no good it runs better with them out. even though it displays the notification a lot, it continues to be playable without optifine Opengl crashes and i have to close minecraft out.

**Comment by tokinGanja** [comment]
optifine's algorithm is the only thing that forces the game to continue through the crash.

**Comment by tokinGanja** [comment]
Not sure what all changed, i do miss my Enhanced portals But i'll cope, Works wonderfully without optifine on the devco servers now no crashes reduced framerate but no issue at all, super playable, Thanks.

**Comment by MJRLegends** [comment]
So this issues fixed?

**Comment by MJRLegends** [comment]
@tokinGanja ???

**Comment by 1John5vs7** [comment]
Just for the record, I am having exactly the same issue in another pack. IntegratedCircuits is throwing exactly the same error: "Shader not supported, disabling aura effect." 

I tried the fix that tokinGanja recommended and it isn't working. I still get the error. It's not until I try moving that I crash though. I take about 2 steps and then BAM, everything freezes.

I've got graphics all turned to minimum. I'm going to try disabling FastCraft next and leaving OptiFine in. I originally started getting the error without ever having installed OptiFine in the first place, just running Fastcraft which comes with the pack by default.
